### PR TITLE
KTOR-686 Add local network address function to engine

### PR DESCRIPTION
**Pull request by rsinukov**
Fixes https://youtrack.jetbrains.com/issue/KTOR-686

### CODE REVIEW

File: ktor-server/ktor-server-cio/jvm/src/io/ktor/server/cio/CIOApplicationEngine.kt [hunk 1]
```diff
@@ -48,6 +49,8 @@ public class CIOApplicationEngine(
     private val startupJob: CompletableDeferred<Unit> = CompletableDeferred()
     private val stopRequest: CompletableJob = Job()
 
+    override val networkAddresses: Deferred<List<NetworkAddress>> = CompletableDeferred()
```
Comment (e5l) [inline ktor-server/ktor-server-cio/jvm/src/io/ktor/server/cio/CIOApplicationEngine.kt:52]: I would add the `suspend` method instead: deferred expose `cancel` and other public calls

**Review by e5l — CHANGES_REQUESTED** [review]

File: ktor-server/ktor-server-cio/jvm/src/io/ktor/server/cio/CIOApplicationEngine.kt (hunk 1 above)
Comment (rsinukov) [reply ktor-server/ktor-server-cio/jvm/src/io/ktor/server/cio/CIOApplicationEngine.kt]: good point, updated

**Review by e5l — APPROVED** [review]
Thanks for the PR, lgtm

**Comment by e5l** [comment]
Please also check the CI

**Pull request merged**